pkgs/top-level/ocaml-packages.nix

```diff
@@ -1836,6 +1871,7 @@ in let inherit (pkgs) callPackage; in rec
   ocamlPackages_latest = ocamlPackages_5_0;
 
   ocamlPackages = ocamlPackages_4_14;
+  ocamlPackages_4_14_janeStreet_0_15 = ocamlPackages_4_14.overrideScope' (self: super: super // super.janeStreet_0_15);
```

Comment:
Is this really needed? It does not look right.

Reply:
I needed this in order to build packages outside of `ocamlPackages` that depend on janestreet v0.15 libraries AND other non-JS OCaml packages, like `stanc`. With `overrideScope'`, any non-JS OCaml modules that depend on JS libraries will pull in v0.15 libraries instead of the default v0.16 libraries. That way, external packages can use both JS and non-JS libraries without accidentally pulling in both v0.15 and v0.16 due to transitive build dependencies. Simply plug in `ocamlPackages_4_14_janeStreet_0_15` as `ocamlPackages` in stanc derivation.

The idea is to override attributes lifted one level up from `janeStreet` (v0.16) by first call to `overrideScope'`, with attributes from `janeStreet_0_15` (v0.15).

Another way of dealing with this is carefully and incrementally crafting transitive dependencies with `.override` to use v0.15, like I had to do for e.g. `bap`. But this seems quite messy.

It's nasty, I agree. I reckon we can get rid of this as soon as v0.16 libraries are supported upstream.
Any cleaner solutions to this are very welcome!

Reply:
Alright. If we agree on the fact that this is a temporary ugly hack, then that’s fine. Let’s hope `stanc` and `ligo` get updated.